targets/psa_switch/externs/psa_hash.cpp

```diff
@@ -0,0 +1,88 @@
+/* Copyright 2021 SYRMIA LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * Dusan Krdzic (dusan.krdzic@syrmia.com)
+ *
+ */
+
+
+#include "psa_hash.h"
+
+namespace {
+
+void
+build_buffer(const std::vector<bm::Field> &fields, bm::ByteContainer &buf) {
+  int nbits = 0;
+  int nbytes;
+  for (const auto &field : fields) {
+    nbits += field.get_nbits();
+  }
+  nbytes = (nbits + 7) / 8;
```

Comment:
since we know the number of bytes needed for storage at this point, could we resize the buffer a single time here? otherwise we potentially allocate memory and move data several times

Reply:
Ok we can merge it like this, and I can take a look later